### DISCUSSION TITLE
On Cache Invalidation - Why is it hard? | /en/2018/06/cache-invalidation/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#1151] [Comments]
https://yihui.org/en/2018/06/cache-invalidation/

[a comment by an automated account is not shown]

**Comment by refeed** [comment]
this is interesting

**Comment by rosedo** [comment]
I agree with everything in the article.

There will be times when you feel the need to cache something but knowing when to invalidate the cache IS hard. You most likely aren't using a cutting-edge database that alerts you when some data has changed. You may not be coding a single-page app that can fetch data at the last moment and therefore you're fetching it too often (and potentially too early to make matters worse).

There are ways to make invalidating cache easier though, like invalidating a whole cache if any random tiny thing is updated anywhere, or by giving the cache a short lifespan with an expiration date.

Some battles are not worth fighting, even if you win.
